src/serac/physics/materials/green_saint_venant_thermoelastic.hpp

```diff
@@ -46,25 +46,24 @@ struct GreenSaintVenantThermoelasticMaterial {
    * @param[in,out] state State variables for this material
    *
    * @return[out] tuple of constitutive outputs. Contains the
-   * Cauchy stress, the volumetric heat capacity in the reference
+   * First Piola stress, the volumetric heat capacity in the reference
    * configuration, the heat generated per unit volume during the time
    * step (units of energy), and the referential heat flux (units of
    * energy per unit time and per unit area).
    */
-  template <typename T1, typename T2, typename T3>
-  auto operator()(State& state, const tensor<T1, 3, 3>& grad_u, T2 theta, const tensor<T3, 3>& grad_theta) const
+  template <typename T1, typename T2, typename T3, int dim>
+  auto operator()(State& state, const tensor<T1, dim, dim>& grad_u, T2 theta, const tensor<T3, dim>& grad_theta) const
   {
     const double K = E / (3.0 * (1.0 - 2.0 * nu));
     const double G = 0.5 * E / (1.0 + nu);
-    static constexpr auto I = Identity<3>();
+    static constexpr auto I = Identity<dim>();
     auto F = grad_u + I;
     const auto Eg = greenStrain(grad_u);
     const auto trEg = tr(Eg);
 
     // stress
     const auto S = 2.0 * G * dev(Eg) + K * (trEg - 3.0 * alpha * (theta - theta_ref)) * I;
```

Comment:
This model used to be hard coded to 3 space dimensions. It's nice that you're making it a template. Some of the "3" literals in these probably come from the 3D derivation. Can you check to see if any of the 3's should be `dim`?